src/main/scala/io/iohk/ethereum/network/KnownNodesManager.scala

```diff
@@ -0,0 +1,88 @@
+package io.iohk.ethereum.network
+
+import java.net.URI
+
+import akka.actor.{Actor, ActorLogging, Props, Scheduler}
+import io.iohk.ethereum.db.storage.KnownNodesStorage
+import io.iohk.ethereum.network.KnownNodesManager.KnownNodesManagerConfig
+
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class KnownNodesManager(
+    config: KnownNodesManagerConfig,
+    knownNodesStorage: KnownNodesStorage,
+    externalSchedulerOpt: Option[Scheduler] = None)
+  extends Actor with ActorLogging {
+
+  import KnownNodesManager._
+
+  private def scheduler = externalSchedulerOpt getOrElse context.system.scheduler
+
+  var knownNodes: Set[URI] = knownNodesStorage.getKnownNodes()
+
+  var toAdd: Set[URI] = Set.empty
+
+  var toRemove: Set[URI] = Set.empty
+
+  scheduler.schedule(config.persistInterval, config.persistInterval) {
+    persistChanges()
```

Comment:
is it safe?
should we send message to ourself here and process message in main actor thread?